PurchasesTests/Purchasing/EntitlementInfosTests.swift

```diff
@@ -956,69 +956,94 @@ class EntitlementInfosTests: XCTestCase {
+        let proCat: Purchases.EntitlementInfo = subscriberInfo.entitlements[expectedEntitlement]!
 
-        expect(proCat.store.rawValue).to(matcher)
+        expect(proCat.store) == expectedStore
     }
 
-    func verifySandbox(_ matcher: Predicate<Bool> = beFalse(), entitlement: String = "pro_cat") {
+    func verifySandbox(_ expectedIsSandbox: Bool = false, expectedEntitlement: String = "pro_cat") {
         let subscriberInfo: Purchases.PurchaserInfo = Purchases.PurchaserInfo(data: response)!
-        let proCat: Purchases.EntitlementInfo = subscriberInfo.entitlements[entitlement]!
+        let proCat: Purchases.EntitlementInfo = subscriberInfo.entitlements[expectedEntitlement]!
 
-        expect(proCat.isSandbox).to(matcher)
+        expect(proCat.isSandbox) == expectedIsSandbox
     }
     
-    func verifyProduct(identifier: Predicate<String> = equal("monthly_freetrial"), entitlement: String = "pro_cat") {
-        verifyProduct(identifier: identifier,
-                latestPurchaseDate: equal(formatter.date(from: "2019-07-26T23:45:40Z")),
-                originalPurchaseDate: equal(formatter.date(from: "2019-07-26T23:30:41Z")),
-                expirationDate: equal(formatter.date(from: "2200-07-26T23:50:40Z")),
-                entitlement: entitlement
+    func verifyProduct(expectedIdentifier: String = "monthly_freetrial", expectedEntitlement: String = "pro_cat") {
+        verifyProduct(expectedIdentifier: expectedIdentifier,
+                      expectedLatestPurchaseDate: formatter.date(from: "2019-07-26T23:45:40Z"),
+                      expectedOriginalPurchaseDate: formatter.date(from: "2019-07-26T23:30:41Z"),
+                      expectedExpirationDate: formatter.date(from: "2200-07-26T23:50:40Z"),
+                      expectedEntitlement: expectedEntitlement
         )
     }
 
-    func verifyProduct(identifier: Predicate<String>,
-                       latestPurchaseDate: Predicate<Date>,
-                       originalPurchaseDate: Predicate<Date>,
-                       expirationDate: Predicate<Date>,
-                       entitlement: String = "pro_cat") {
+    func verifyProduct(expectedIdentifier: String,
+                       expectedLatestPurchaseDate: Date?,
+                       expectedOriginalPurchaseDate: Date?,
+                       expectedExpirationDate: Date?,
+                       expectedEntitlement: String = "pro_cat") {
         let subscriberInfo: Purchases.PurchaserInfo = Purchases.PurchaserInfo(data: response)!
-        let proCat: Purchases.EntitlementInfo = subscriberInfo.entitlements[entitlement]!
+        let proCat: Purchases.EntitlementInfo = subscriberInfo.entitlements[expectedEntitlement]!
+
+        if (expectedLatestPurchaseDate != nil) {
+            expect(proCat.latestPurchaseDate) == expectedLatestPurchaseDate
+        } else {
+            expect(proCat.latestPurchaseDate).to(beNil())
+        }
```

Comment:
I've never actually double-checked, maybe the overloaded `==` operator from Quick/Nimble does actually work correctly with nils?